Source/Core/DolphinQt/Debugger/CodeDiffProxyModel.h

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <optional>
+
+#include <QSortFilterProxyModel>
+#include <QString>
+
+#include "Common/CommonTypes.h"
+
+namespace Core
+{
+class BranchWatch;
+}
+class CodeDiffDialog;
+class CodeDiffTableModel;
+class QGridLayout;
+
+class CodeDiffProxyModel final : public QSortFilterProxyModel
+{
+  Q_OBJECT
+
+  friend CodeDiffDialog;
+
+public:
+  explicit CodeDiffProxyModel(const Core::BranchWatch& branch_watch, QObject* parent = nullptr)
+      : QSortFilterProxyModel(parent), m_branch_watch(branch_watch)
+  {
+  }
+  CodeDiffTableModel* sourceModel() const;
+  bool filterAcceptsRow(int source_row, const QModelIndex& source_parent) const override;
+
+  template <bool CodeDiffProxyModel::*member>
+  void OnToggled(bool enabled)
+  {
+    this->*member = enabled;
+    invalidateRowsFilter();
+  }
+  void OnSymbolTextChanged(const QString& text)
```

Comment:
```suggestion
  void OnSymbolTextChanged(QString text)
```

and std::move into `m_symbol_name`

Reply:
This method is a slot for the signal, `QLineEdit::textChanged(const QString& text)`, and isn't needed for anything else. I would rather keep the signature matching, as it makes a lambda in `CodeDiffDialog::CreateLayout()` simple to reuse.